// @frank-mayer/react-tag-cloud/main.js
var $g5qWx$TagCloud = require("TagCloud");
var $g5qWx$react = require("react");

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$defineInteropFlag(a) {
  Object.defineProperty(a, '__esModule', {value: true, configurable: true});
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

$parcel$defineInteropFlag(module.exports);

$parcel$export(module.exports, "TagCloud", () => $f5bfd4ce37214f4f$export$61a620005b8c73bd);
$parcel$export(module.exports, "default", () => $f5bfd4ce37214f4f$export$2e2bcd8739ae039);


const $f5bfd4ce37214f4f$var$getIsInitialized = (id)=>{
    const key = "__tag-cloud-" + id;
    if (key in window) return Boolean(window[key]);
    return window[key] = false;
};
const $f5bfd4ce37214f4f$var$setIsInitialized = (id, value)=>{
    const key = "__tag-cloud-" + id;
    window[key] = value;
};
const $f5bfd4ce37214f4f$export$61a620005b8c73bd = (props)=>{
    const ref = (0, $g5qWx$react.useRef)(null);
    const key = [
        props.id,
        props.className,
        JSON.stringify(props.children)
    ].join("-");
    (0, $g5qWx$react.useEffect)(()=>{
        if ($f5bfd4ce37214f4f$var$getIsInitialized(key) || !ref.current) return;
        if (props.children.length === 0) {
            console.error("TagCloud: No children provided.");
            return;
        }
        const options = props.options ? typeof props.options == "function" ? props.options(window) : props.options : {};
        const tagCloud = (0, ($parcel$interopDefault($g5qWx$TagCloud)))(ref.current, props.children, options);
        $f5bfd4ce37214f4f$var$setIsInitialized(key, true);
        if (props.onClick) {
            const elements = Array.from(ref.current.getElementsByClassName(options.itemClass ?? "tagcloud--item"));
            if (props.onClick) for (const el of elements)el.addEventListener("click", (event)=>{
                props.onClick(el.innerText, event);
            }, props.onClickOptions);
        }
        return ()=>{
            try {
                tagCloud?.destroy();
            } finally{
                $f5bfd4ce37214f4f$var$setIsInitialized(key, false);
            }
        };
    }, [
        ref,
        key,
        props
    ]);
    return (0, $g5qWx$react.createElement)("div", {
        id: props.id,
        className: props.className,
        style: props.style,
        ref: ref
    });
};
var $f5bfd4ce37214f4f$export$2e2bcd8739ae039 = $f5bfd4ce37214f4f$export$61a620005b8c73bd;
